101-route-server/azuredeploy.json

```diff
@@ -0,0 +1,151 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vnetName": {
+      "type": "string",
+      "defaultValue": "routeservervnet",
+      "metadata": {
+        "description": "Name of new or existing vnet to which Azure Route Server should be deployed."
+      }
+    },
+    "vnetIpPrefix": {
+      "type": "string",
+      "defaultValue": "10.1.0.0/16",
+      "metadata": {
+          "description": "IP prefix for available addresses in vnet address space."
+      }
+    },
+    "vnetNew-or-Existing": {
+        "type": "string",
+        "allowedValues": [
+            "New",
+            "Existing"
+        ],
+        "defaultValue": "New",
+        "metadata": {
+            "description": "Specify whether to provision new vnet or deploy to existing vnet."
+        }
+    },
+    "routeServerSubnetIpPrefix": {
+      "type": "string",
+      "defaultValue": "10.1.1.0/27",
+      "metadata": {
+          "description": "Route Server subnet IP prefix MUST be within vnet IP prefix address space."
+      }
+    },
+    "firstRouteServerName": {
+      "type": "string",
+      "defaultValue": "routeserver",
+      "metadata": {
+        "description": "Name of Route Server."
+      }
+    },
+    "routeServerBgpConnectionName": {
+      "type": "string",
+      "defaultValue": "conn1",
+      "metadata": {
+        "description": "Name of BGP connection."
+      }
+    },
+    "peerAsn": {
+      "type": "string",
+      "defaultValue": "65002",
+      "metadata": {
+        "description": "Peer ASN connecting to."
+      }
+    },
+    "peerIp": {
+      "type": "string",
+      "defaultValue": "10.0.1.4",
+      "metadata": {
+        "description": "Peer IP connecting to."
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Azure region for Route Server and virtual network."
+      }
+    }    
+  },
+  "variables": {    
+    "ipconfigName": "[concat(parameters('firstRouteServerName'), '/ipconfig1')]",
+    "bgpConnectionName": "[concat(parameters('firstRouteServerName'), '/', parameters('routeServerBgpConnectionName'))]",
+    "routeserver-subnet-name": "RouteServerSubnet"
+  },
+  "resources": [
+      {
+        "condition": "[equals(parameters('vnetNew-or-Existing'), 'New')]",
+        "apiVersion": "2020-05-01",
+        "name": "[parameters('vnetName')]",
+        "type": "Microsoft.Network/virtualNetworks",
+        "location": "[parameters('location')]",
+        "properties": {
+            "addressSpace": {
+                "addressPrefixes": [
+                    "[parameters('vnetIpPrefix')]"
+                ]
+            },
+            "subnets": [
```

Comment:
remove this and then remove the condition on the subnet below since it's always deployed

Reply:
This subnet creation has been removed.